src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompilationInfo.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Text.RegularExpressions
+{
+    public class RegexCompilationInfo
+    {
+        private string _pattern;
+        private string _name;
+        private string _nspace;
+        private TimeSpan _matchTimeout;
+
+        public RegexCompilationInfo(string pattern, RegexOptions options, string name, string fullnamespace, bool ispublic)
+            : this(pattern, options, name, fullnamespace, ispublic, Regex.DefaultMatchTimeout)
+        {
+        }
+
+        public RegexCompilationInfo(string pattern, RegexOptions options, string name, string fullnamespace, bool ispublic, TimeSpan matchTimeout)
+        {
+            Pattern = pattern;
+            Name = name;
+            Namespace = fullnamespace;
+            Options = options;
+            IsPublic = ispublic;
+            MatchTimeout = matchTimeout;
+        }
+
+        public string Pattern
+        {
+            get => _pattern;
+            set => _pattern = value ?? throw new ArgumentNullException("value");
```

Comment:
Nit: `nameof(value)`

Reply:
or `nameof(Pattern)` (and same for others)? I see desktop used `"value"` but I can't find other examples in our code where we use `value` (which only the implementation sees) instead of the actual property name (wihch the caller sees). Admittedly I only looked briefly. @stephentoub do we have a convention?